Mage/src/main/java/mage/abilities/effects/mana/BasicManaEffect.java

```diff
@@ -39,7 +39,7 @@ public BasicManaEffect(ConditionalMana conditionalMana, DynamicValue netAmount)
 
     public BasicManaEffect(final BasicManaEffect effect) {
         super(effect);
-        this.manaTemplate = effect.manaTemplate.copy();
```

Comment:
Must add comments why you don't use copy code for the field. It's useful for devs. Miss copy in the constructor is a popular bug.

P.S. After some research -- `manaTemplate` field is forwarded further to outside classes (see screenshot with netMana). So it can be potentially changed later. I recommends to add standard copy code here to protect it from such bugs in the future.

![shot_221010_062919](https://user-images.githubusercontent.com/8344157/194791871-e5d0f457-d8b3-45d0-9387-aa2997e50ead.png)

Reply:
I've finished cleaning this up so that getManaTemplate returns a copy, allowing copy to be removed in all other locations. Removed a few more redundant assignments and made fields final. Comments added.